### DISCUSSION TITLE
set type hint Optional[List[str]],  but receive value as ['a,b,c'], not ['a', 'b', 'c']

### DISCUSSION THREAD

**Discussion by shizidushu** [fastapi/fastapi#6727] [Questions] [answered]
There is the steps:

- `app/main.py`
```python
from typing import Optional
from fastapi import FastAPI, File, UploadFile, Form

from typing import List

app = FastAPI()

@app.post("/")
async def root(category: Optional[List[str]] = Form(..., title="category"), file: UploadFile = File(..., title="uploaded file")):
    params = locals()
    return {"params": repr(params)}
```
- Dockerfile
```
FROM tiangolo/uvicorn-gunicorn-fastapi:python3.8

RUN pip3 install python-multipart

COPY ./app /app
```
- Build
```
docker build -t fastapi-test ./
```
- Run
```
docker run --rm --name fastapi-test-form -p 8008:80 fastapi-test
```

- Try

![图片](https://user-images.githubusercontent.com/13694980/111511433-b2810000-8789-11eb-84f9-f4a14152fbff.png)

I'd like to get `category=['a', 'b', 'c']`, but not.
I can't reproduce it when using my windows pc.

**Comment by hellocoldworld** [comment, marked as answer]
You're passing 'a,b,c' in the same category Form "field". You would need to pass the three letter in three distinct form fields, all of them named 'category'. For example, the following cURL should work

 ```bash
$ curl -X 'POST' \
>   'http://127.0.0.1:8000/' \
>   -H 'accept: application/json' \
>   -H 'Content-Type: multipart/form-data' \
>   -F 'category=a' \
>   -F 'category=b' \
>   -F 'category=c' \
>   -F 'file=@check.html;type=text/html'
 ```

**Comment by shizidushu** [comment]
@hellocoldworld Thanks. I should change [the way to send the request](https://stackoverflow.com/questions/65335559/sending-array-data-with-multipart-form-data-post-request-in-axios-vue-js).

**Comment by tiangolo** [comment]
Thanks for the help here @hellocoldworld ! 👏  🙇 

Thanks for reporting back and closing the issue @shizidushu  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
